apiserver/facades/client/application/application.go

```diff
@@ -1336,8 +1344,68 @@ func (api *APIBase) SetRelationsSuspended(ctx context.Context, args params.Relat
 // Consume adds remote applications to the model without creating any
 // relations.
 func (api *APIBase) Consume(ctx context.Context, args params.ConsumeApplicationArgsV5) (params.ErrorResults, error) {
-	return params.ErrorResults{}, internalerrors.Errorf("cross model relations are disabled until "+
-		"backend functionality is moved to domain: %w", errors.NotImplemented)
+	if err := api.checkCanWrite(ctx); err != nil {
+		return params.ErrorResults{}, errors.Trace(err)
+	}
+	if err := api.check.ChangeAllowed(ctx); err != nil {
+		return params.ErrorResults{}, errors.Trace(err)
+	}
+
+	results := make([]params.ErrorResult, len(args.Args))
+	for i, arg := range args.Args {
+		err := api.consumeOne(ctx, arg)
+		results[i].Error = apiservererrors.ServerError(err)
+	}
+	return params.ErrorResults{
+		Results: results,
+	}, nil
+}
+
+func (api *APIBase) consumeOne(ctx context.Context, arg params.ConsumeApplicationArgV5) error {
+	sourceModelTag, err := names.ParseModelTag(arg.SourceModelTag)
+	if err != nil {
+		return internalerrors.Errorf("parsing source model tag: %w", err).Add(coreerrors.BadRequest)
+	}
+
+	// Maybe save the details of the controller hosting the offer.
+	var externalControllerTag names.ControllerTag
+	if arg.ControllerInfo != nil {
+		externalControllerTag, err = api.saveExternalController(ctx, *arg.ControllerInfo, sourceModelTag)
+		if err != nil {
+			return internalerrors.Errorf("saving external controller info: %w", err)
+		}
+	}
+
+	_ = externalControllerTag
```

Comment:
Why? Are you planning to use it later? Otherwise the result from saveExternalController could be ignored right?

Reply:
I am. It's in a follow up.